EventFilter/EcalRawToDigi/plugins/EcalRawToDigiGPU.cc

```diff
@@ -92,17 +87,30 @@ void EcalRawToDigiGPU::acquire(edm::Event const& event,
   // scratch
   ecal::raw::ScratchDataGPU scratchGPU = {cms::cuda::make_device_unique<uint32_t[]>(2, ctx.stream())};
 
+  // make a first iteration over the FEDs to compute the total buffer size
+  uint32_t size = 0;
+  uint32_t feds = 0;
+  for (auto const& fed : fedsToUnpack_) {
+    auto const& data = rawDataHandle->FEDData(fed);
+    auto const nbytes = data.size();
+
+    // skip empty FEDs
+    if (nbytes < globalFieds::EMPTYEVENTSIZE)
+      continue;
```

Comment:
How about filling a `std::vector<int> nonEmptyFeds;` here with the FEDs that are not empty and loop over that one for the second loop in line 127?

Reply:
Do you have many empty FEDs in ECAL ?
If most are non-empty, using an extra intermediate vector would likely be (marginally) slower.

Reply:
In normal PU conditions I expect most FEDs to be not empty. So keeping the same vector is probably better.